include/zephyr/drivers/battery.h

```diff
@@ -0,0 +1,200 @@
+/*
+ * Copyright 2022 Google LLC
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#ifdef CONFIG_BATTERY
+
+#ifndef ZEPHYR_INCLUDE_DRIVERS_BATTERY_H_
+#define ZEPHYR_INCLUDE_DRIVERS_BATTERY_H_
+
+#include <stdbool.h>
+#include <stdint.h>
+#include <zephyr/device.h>
+
+/*
+ * Sometimes we have hardware to detect battery present, sometimes we have to
+ * wait until we've been able to talk to the battery.
+ */
+enum battery_present {
+	BATTERY_PRESENCE_UNINITIALIZED = -1,
+	BATTERY_PRESENCE_NO = 0,
+	BATTERY_PRESENCE_YES = 1,
+	BATTERY_PRESENCE_UNKNOWN,
+};
+
+enum battery_disconnect_state {
+	BATTERY_DISCONNECTED = 0,
+	BATTERY_NOT_DISCONNECTED,
+	BATTERY_DISCONNECT_ERROR,
```

Comment:
Better to name it "battery_connect_state" and have only 2 values "connected" and "disconnected". Since this is mostly a GPIO, not sure what "battery_disconnect_error" can be.